tests/e2e/utils/rds-s3/auto-rds-s3-setup.py

```diff
@@ -98,6 +101,75 @@ def setup_s3(s3_client, secrets_manager_client):
     setup_s3_bucket(s3_client)
     if CREDENTIALS_OPTION == "static":
         setup_s3_secrets(secrets_manager_client)
+        return None
+    else:
+        return setup_pipeline_irsa()
+
+
+def setup_pipeline_irsa():
+    print_banner("Create OIDC IAM role for Pipelines")
+    iam_client = get_iam_client(region=CLUSTER_REGION)
+    PIPELINE_OIDC_ROLE_NAME_PREFIX = "kf-pipeline-role"
+
+    pipeline_oidc_role_name = f"{PIPELINE_OIDC_ROLE_NAME_PREFIX}-{CLUSTER_NAME}"[:64]
+    try:
+        create_pipeline_oidc_role(pipeline_oidc_role_name, iam_client)
+    except Exception as e:
+        print(e)
+
+    oidc_role_arn = get_role_arn(iam_client, pipeline_oidc_role_name)
+    return oidc_role_arn
+
+
+def profile_trust_policy(account_id):
+    eks_client = get_eks_client(region=CLUSTER_REGION)
+
+    resp = eks_client.describe_cluster(name=CLUSTER_NAME)
+    oidc_url = resp["cluster"]["identity"]["oidc"]["issuer"].split("https://")[1]
+
+    trust_policy = {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Effect": "Allow",
+                "Principal": {
+                    "Federated": f"arn:aws:iam::{account_id}:oidc-provider/{oidc_url}"
+                },
+                "Action": "sts:AssumeRoleWithWebIdentity",
+                "Condition": {
+                    "StringEquals": {
+                        f"{oidc_url}:aud": "sts.amazonaws.com",
+                        f"{oidc_url}:sub": [
+                            "system:serviceaccount:kubeflow:ml-pipeline",
+                            "system:serviceaccount:kubeflow-user-example-com:default-editor",
```

Comment:
this works but not be a good practice usage wise because users might add more permissions for their profile and the pipeline backend pod will also get those permissions

can we create separate roles for user and the backend?

Reply:
created seperate frontend and backend roles